FirebaseAuth/Sources/Auth/FIRAuth.m

```diff
@@ -485,85 +491,124 @@ - (nullable instancetype)initWithAPIKey:(NSString *)APIKey
     _firebaseAppName = [appName copy];
 #if TARGET_OS_IOS
     _settings = [[FIRAuthSettings alloc] init];
+
+    [GULAppDelegateSwizzler proxyOriginalDelegateIncludingAPNSMethods];
+    [GULSceneDelegateSwizzler proxyOriginalSceneDelegate];
+#endif  // TARGET_OS_IOS
+
+#if TARGET_OS_IOS || TARGET_OS_TV || TARGET_OS_MACCATALYST
     static Class applicationClass = nil;
     // iOS App extensions should not call [UIApplication sharedApplication], even if UIApplication
     // responds to it.
     if (![GULAppEnvironmentUtil isAppExtension]) {
       Class cls = NSClassFromString(@"UIApplication");
-      if (cls && [cls respondsToSelector:NSSelectorFromString(@"sharedApplication")]) {
+      if (cls && [cls respondsToSelector:@selector(sharedApplication)]) {
```

Comment:
I'm curious if there's a reason for switching away from `NSSelectorFromString` here

Reply:
No big reason -- is one preferred over the other? I see more instances of `@selector` in the Firebase Auth codebase than `NSSelectorFromString`, so I went with `@selector` (and it seems less error-prone, since it doesn't use a string-match), but can definitely change it back.